src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/InputAllowNonOverlappingLocalVars.java

```diff
@@ -0,0 +1,49 @@
+/*
+Input test file for RequireThisCheck.
+Created: 2017
+ */
+
+package com.puppycrawl.tools.checkstyle.checks.coding;
+
+class InputAllowNonOverlappingLocalVars {
+
+    private String s1 = "foo1";
+    String s2 = "foo2";
+
+    InputAllowNonOverlappingLocalVars() {
+        s1 = "bar1"; // Violation. Requires "this".
+        String s2;
+        s2 = "bar2"; // No violation. Non-overlapping overshadow by local var allowed.
```

Comment:
Check's code uses `overlapping` at line 480 for a method name.
Where did `overshadowing` come from? We should similar naming conventions between code and documentation. If we don't then, imo, it is clear that one area's convention is wrong and should be changed.